src/api/topics/topics.service.ts

```diff
@@ -0,0 +1,44 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository, UpdateResult } from 'typeorm';
+
+import { CreateTopicDto } from '@api/topics/dto/create-topic.dto';
+import { UpdateTopicDto } from '@api/topics/dto/update-topic.dto';
+import { TopicEntity } from '@api/topics/topic.entity';
+
+@Injectable()
+export class TopicsService {
+  constructor(
+    @InjectRepository(TopicEntity)
+    private topicRepository: Repository<TopicEntity>,
+  ) {}
+
+  create(createTopicDto: CreateTopicDto): Promise<TopicEntity> {
+    return this.topicRepository.save(createTopicDto);
+  }
+
+  findAll(): Promise<TopicEntity[]> {
+    return this.topicRepository.find();
+  }
+
+  findOne(id: number): Promise<TopicEntity | null> {
+    return this.topicRepository.findOne({
```

Comment:
Here, you could use "findOneOrFail" and then handle the error. So, when you reuse this method, the error is always caught. For example:

  async findById(id: number): Promise<TopicEntity> {
    try {
      const topic = await this.topicRepository.findOneOrFail({ id });
      return topic;
    } catch (error) {
      throw new TopicNotFoundException(id);
    }
  }
  
  Then in update:
  
  async update(
    id: number,
    updateTopicDto: UpdateTopicDto,
  ): Promise<UpdateResult> {
    const topic = await this.findById=d(id);
    try {
      return this.topicRepository.update({ id }, updateTopicDto);
      return topic;
    } catch (error) {
      throw new TopicNotUpdateException(id, error);
    }
  }
  
  This way each method handles its errors and the code is cleaner, I even suggest changing the name of findOne to findById to make it more descriptive.

What do you think?